src/features/order/orderController.ts

```diff
@@ -127,22 +94,25 @@ const getAllOrders = async (req: Request, res: Response, next: NextFunction) =>
     }
 
     const orders = await orderService.getAllOrders();
-    res.status(200).json(orders);
+    res.status(200).json("Orders retrieved successfully");
   } catch (error) {
     next(error);
   }
 };
 
 const getVendorProductsOnOrders = async (req: Request, res: Response, next: NextFunction) => {
   try {
-    if (!req.user || req.user.role !== "vendor") {
+    const { vendorId } = req.params;
+    const userRole = req.user?.role;
+
+    // Allow both admins and vendors to access this route
+    if (!req.user || (userRole !== "vendor" && userRole !== "admin")) {
```

Comment:
@Simply-Aidan use roleMiddleware to allow the appropraite role to access the route. you will find the roleMiddleware in the code